addons/misra.py

```diff
@@ -4110,7 +4126,7 @@ def loadRuleTexts(self, filename):
         expect_more = False
 
         Rule_pattern = re.compile(r'^Rule ([0-9]+).([0-9]+)')
-        severity_pattern = re.compile(r'.*[ ]*(Advisory|Required|Mandatory)$')
+        severity_pattern = re.compile(r'^.* *(Advisory|Required|Mandatory).*$')
```

Comment:
This bugfix is not related to the initial scope of this PR.  
The regex did not handle trailing whitespaces in the rule files. Since this error was not caught, I preferred to fix it.

Reply:
do you only want to allow trailing white spaces.. not other stuff.. then the pattern can be more strict. I think `.* *` looks weird that is the same as `.*` isn't it?

Reply:
You are right! My bad, I wanted to accept white spaces indeed, not any characters. I've fixed it.